### ISSUE TITLE
Failed to mock interface in Debian but succeeded in Mac OS

### ISSUE THREAD

**Issue by pancerZH** [mockito/mockito#2191]
JDK: JDK11
Mockito Version: 3.5.11

Hi Mockito Team:

I am using Mockito as my unit test framework, but currently I am experiencing a really wired issue:
1. Some unit tests, which include mocks for interfaces or abstract classes, failed in our CI/CD process, which was a Linux. But these tests passed on my MacBook Pro.
2. I built a docker container to reproduce these issues, and it did do it.
3. Some details:
    - The first time I ran unit tests in the container, some tests failed; But if I ran unit tests the second time, all the failed tests passed.
    - If I ran the unit tests individually, or package by package, these tests passed; But they failed when run as a whole.
    - After I replaced all the mock for interfaces with mock for implemented classes, all the tests passed.
    - Addition: when the unit tests containing mock for interfaces failed, a NullPointerException would be thrown, but some other unit tests were also influenced, though they did not contain mock for interfaces: Exceptions like InValidUseOfMock appeared.

I have to say this is really wired and it costed me a lot of time to fix it, but still could not find the root cause. Even a simple code like:
```java
Map<String, String> map = mock(Map.class);
map.add("test", "test");
```
would also fail with a NullPointerException of the second line.

Could you have look on it? Thank you for your help!

**Comment by ostecke** [comment]
FYI: I had similar issues, and our fix was to remove code (from another test) that mocks Method.class. Please see https://github.com/mockito/mockito/issues/2026 for a discussion on this.

**Comment by pancerZH** [comment]
The final solution for me is to change those beans injected by interfaces or abstract classes to concrete classes to avoid mocking interfaces or abstract classes.

**Comment by TimvdLippe** [comment]
Unfortunately, `Map` is one of the classes that Mockito relies on internally for its behavior. Stubbing `Map` will therefore lead to undefined behavior. Additionally, it is advised not to mock classes you don't own: https://github.com/mockito/mockito/wiki/How-to-write-good-tests#dont-mock-a-type-you-dont-own We are working on improving the user experience by working on a `DoNotMock` feature to avoid mocking classes/methods that are known to crash Mockito internals (#1833). Therefore, I am closing this as "Infeasible". Apologies for the uninformative exception that is thrown.

**Comment by pancerZH** [comment]
> Unfortunately, `Map` is one of the classes that Mockito relies on internally for its behavior. Stubbing `Map` will therefore lead to undefined behavior. Additionally, it is advised not to mock classes you don't own: https://github.com/mockito/mockito/wiki/How-to-write-good-tests#dont-mock-a-type-you-dont-own We are working on improving the user experience by working on a `DoNotMock` feature to avoid mocking classes/methods that are known to crash Mockito internals (#1833). Therefore, I am closing this as "Infeasible". Apologies for the uninformative exception that is thrown.

Thank you for your reply, and I want to confirm that the `Map` here is only an example, the key point is the failure of mock of injected beans with interface or abstract class type.

**Comment by TimvdLippe** [comment]
Oh apologies for that. I will reopen, but I think we are going to need a standalone reproduction case for us to investigate. It could potentially be test flakiness that you are running into, which is outside of Mockito's scope.

**Comment by pancerZH** [comment]
> Oh apologies for that. I will reopen, but I think we are going to need a standalone reproduction case for us to investigate. It could potentially be test flakiness that you are running into, which is outside of Mockito's scope.

Yeah...but I could not provided the project or make a detailed description, which involves complicated dependencies and a lot of other things. I am wondering whether this has something to do with the lifecycle of Spring, Mockito and JUnit? Because in Linux environment, it would pass/fail randomly the first time, but always pass the second time I run `gradle test`. It looks like somthing about building cache.
